From 7533ad492b3c7e9fd3901ab948eb6a030761ce2f Mon Sep 17 00:00:00 2001
From: Alan Jenkins <alan.christopher.jenkins@gmail.com>
Date: Mon, 25 Sep 2017 11:45:03 +0100
Subject: [PATCH] units: add missing ordering deps for Conflicts= of
 emergency.service

1. If we exited emergency mode immediately, we don't want to have an
   irreversible stop job still running for syslog.socket.  I _suspect_ that
   can't happen, but let's not waste effort working out exactly why it's
   impossible and not just very improbable.

2. Similarly, it seems undesirable to have rescue.service and
   emergency.service both running with an open FD of /dev/console, for
   however short a period.

(cherry picked from commit 6f6d1a8a6abae490e14e6a448895e36c6cefdefc)
---
 units/emergency.service.in | 1 +
 units/syslog.socket        | 1 +
 2 files changed, 2 insertions(+)

diff --git a/units/emergency.service.in b/units/emergency.service.in
index 8768fe137e..27c9a1c23e 100644
--- a/units/emergency.service.in
+++ b/units/emergency.service.in
@@ -12,6 +12,7 @@ DefaultDependencies=no
 Conflicts=shutdown.target
 Conflicts=rescue.service
 Before=shutdown.target
+Before=rescue.service
 
 [Service]
 Environment=HOME=/root
diff --git a/units/syslog.socket b/units/syslog.socket
index 372e8fcd45..43981904ea 100644
--- a/units/syslog.socket
+++ b/units/syslog.socket
@@ -18,6 +18,7 @@ Before=shutdown.target
 
 # Don't try to activate syslog.service if sysinit.target has failed.
 Conflicts=emergency.service
+Before=emergency.service
 
 [Socket]
 ListenDatagram=/run/systemd/journal/syslog
